lib/ansible/modules/system/pamd.py

```diff
@@ -686,7 +742,28 @@ def __str__(self):
 
         lines.insert(1, "# Updated by Ansible - " + datetime.now().isoformat())
 
-        return '\n'.join(lines)
+        return '\n'.join(lines) + '\n'
+
+
+def parse_module_arguments(module_arguments):
+    # Return empty list if we have no args to parse
+    if not module_arguments:
+        return []
+    elif not module_arguments[0]:
```

Comment:
What does this check for?

Reply:
When `parse_module_arguments` is used from within the `PamdRule` class to parse out the arguments in an existing rule in a pamd file it can potentially send nothing if there are no matches. 

This checks for the cases where the passed in args are `None`, and empty list `[]` and the second one checks for a list with a single empty item. i.e `['']` and then returns early.

Reply:
looks like it checks that the first element is empty, but not that there is a single element...

Reply:
The first part of the if handles module_arguments is:
- an empty string
- an empty list
- None

The second part of the if handles module_arguments is:
- a list with a single empty item ['']

I can add a check like `isinstance(module_arguments, list) and len(module_arguments) == 1 and not module_arguments[0]` if that is preferable.

Reply:
Yes, and that would narrow it down to one `if`statement.

Reply:
Yeah that's probably more defensive given `UserError`, I _could_ be possible, though probably not that likely that there would be some array `[None,'foo', 'bar']`

Reply:
I don't think it narrows it down to a single `if` since `module_arguments` can be an empty string or `None` and if it is we still want to `return []` which the first `if` covers. My change for this would look like

```python
    if not module_arguments:
        return []
    elif isinstance(module_arguments, list) and len(module_arguments) == 1 and not module_arguments[0]:
        return []
```

Reply:
Ack. Looks like a good change.

Reply:
I mean, you could do an `or` but that might break pep8 on line length, I dont care either way, the major thing for me was just being defensive about index 0 potentially being a `None` value AND not the _only_ index.

Reply:
Yeah my preference is for shorter/simpler if tests. I'll update this. I wonder if I can make a suggested change to myself and apply it hehe.